src/NzbDrone.Core/Organizer/FileNameBuilder.cs

```diff
@@ -280,6 +281,37 @@ public static string CleanFolderName(string name)
             return name.Trim(' ', '.');
         }
 
+        public bool RequiresEpisodeTitle(Series series, List<Episode> episodes)
```

Comment:
Encapsulate this in a cache, it's going to be called a lot and you don't want the same regex to be evaluated over and over again.
When config is changed, pull all formats, check which ones have Episode Title and store the resulting 3 booleans in the cache.

Reply:
Added the cache, based on the format string, it could lead to more than 3 values stored, but only if the user is changing formats and that seems like an unlikely use case to worry about.

Reply:
That's fine. Not like the user is going to have hundreds of patterns and never reboot.